### PR TITLE
Remove custom `provided` scope in favor of `compileOnly`.

### PR DESCRIPTION
**Pull request by xian**
We need to make sure `pom.xml`s and javadoc aren't deleteriously altered by this change.

### CODE REVIEW

File: robolectric-resources/build.gradle [hunk 1]
```diff
@@ -6,8 +6,8 @@ dependencies {
     // Compile dependencies
     compile "com.ximpleware:vtd-xml:2.11"
     compile "com.google.guava:guava:19.0"
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
-    provided "com.intellij:annotations:12.0"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-resources/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true } ?

File: robolectric-shadows/shadows-maps/build.gradle [hunk 2]
```diff
@@ -8,10 +8,10 @@ dependencies {
     compile project(":robolectric")
     compile project(":robolectric-shadows/shadows-core")
 
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-shadows/shadows-maps/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric-shadows/shadows-multidex/build.gradle [hunk 3]
```diff
@@ -8,9 +8,9 @@ dependencies {
     compile project(":robolectric")
     compile project(":robolectric-shadows/shadows-core")
 
-    provided "com.android.support:multidex:1.0.1"
+    compileOnly "com.android.support:multidex:1.0.1"
 
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-shadows/shadows-multidex/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric-shadows/shadows-play-services/build.gradle [hunk 4]
```diff
@@ -8,11 +8,11 @@ dependencies {
...
 
-    provided "com.android.support:support-v4:23.2.0"
-    provided "com.google.android.gms:play-services-base:8.4.0"
-    provided "com.google.android.gms:play-services-basement:8.4.0"
+    compileOnly "com.android.support:support-v4:23.2.0"
+    compileOnly "com.google.android.gms:play-services-base:8.4.0"
+    compileOnly "com.google.android.gms:play-services-basement:8.4.0"
 
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-shadows/shadows-play-services/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric-shadows/shadows-support-v4/build.gradle [hunk 5]
```diff
@@ -14,9 +14,9 @@ dependencies {
     compile project(":robolectric-shadows/shadows-core")
 
     // Compile dependencies
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
-    provided "com.android.support:support-v4:23.2.0"
-    provided "com.android.support:internal_impl:23.2.0"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-shadows/shadows-support-v4/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric-utils/build.gradle [hunk 6]
```diff
@@ -1,8 +1,8 @@
 dependencies {
     // Compile dependencies
-    provided "junit:junit:4.8.2"
-    provided "org.hamcrest:hamcrest-core:1.3"
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
+    compileOnly "junit:junit:4.8.2"
+    compileOnly "org.hamcrest:hamcrest-core:1.3"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric-utils/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric/build.gradle [hunk 7]
```diff
@@ -31,18 +31,19 @@ dependencies {
...
-    provided "com.intellij:annotations:12.0"
+    compileOnly "com.intellij:annotations:12.0"
 
     compile "org.apache.ant:ant:1.8.0"
     compile("org.apache.maven:maven-ant-tasks:2.1.3") {
         exclude group: "junit", module: "junit"
     }
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
-    provided "junit:junit:4.8.2"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

File: robolectric/build.gradle [hunk 8]
```diff
@@ -31,18 +31,19 @@ dependencies {
...
+    compileOnly "junit:junit:4.8.2"
 
     // Testing dependencies
     testCompile "junit:junit:4.8.2"
     testCompile "org.hamcrest:hamcrest-core:1.3"
     testCompile "org.assertj:assertj-core:2.0.0"
     testCompile "org.mockito:mockito-core:1.8.0"
+    testRuntime "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```
Comment (jongerrish) [inline robolectric/build.gradle]: ("org.robolectric:android-all:${AndroidSdk.versions[AndroidSdk.latestVersion]}") { force = true }

Comment (jongerrish) [inline robolectric/build.gradle]: Nice if we can get rid of that custom rule :-) do you also want to remove buildSrc/src/main/groovy/provided.groovy ?

File: robolectric-resources/build.gradle (hunk 1 above)
Comment (xian) [inline robolectric-resources/build.gradle]: See the last commit on https://github.com/robolectric/robolectric/tree/api_24-jdk7-compat, I've refactored the `AndroidSdk` stuff there.

**Pull request merged**